### PR TITLE
Kali Linux platform detection support

### PR DESCRIPTION
**Pull request by mattray**
Add support for detecting [Kali Linux](https://www.kali.org)

Signed-off-by: Matt Ray <github@mattray.dev>

### CODE REVIEW

**Comment by mattray** [comment]
FWIW I installed Chef via the .DEB on the Kali box and ohai works just fine. They will both report the same platform, family and version after this.

**Comment by mattray** [comment]
Validated that this works with an aarch64 Kali install too

File: lib/train/platforms/detect/specifications/os.rb [hunk 1]
```diff
@@ -76,6 +76,13 @@ def self.load
             true
           end
         end
+      plat.name("kali").title("Kali Linux").in_family("debian")
+        .detect do
+          if linux_os_release && linux_os_release["ID"].eql?("kali")
+            @platform[:release] = linux_os_release["VERSION"]
```
Comment (zenspider) [inline lib/train/platforms/detect/specifications/os.rb]: This is 3 calls to `linux_os_release`... please use local variables to make it only 1.

File: lib/train/platforms/detect/specifications/os.rb [hunk 2]
```diff
@@ -76,6 +76,13 @@ def self.load
             true
           end
         end
+      plat.name("kali").title("Kali Linux").in_family("debian")
+        .detect do
+          if linux_os_release && linux_os_release["ID"].eql?("kali")
```
Comment (zenspider) [inline lib/train/platforms/detect/specifications/os.rb]: `eql?` is _never_ the right comparator to call and can't succeed in the wild. Please stick with `==`.

File: test/unit/platforms/os_detect_test.rb [hunk 3]
```diff
@@ -202,7 +216,7 @@ def debian_scan(id, version)
 
   describe "/etc/os-release" do
     describe "when not on a wrlinux build" do
-      it "fail back to genaric linux" do
+      it "fail back to generic linux" do
```
Comment (zenspider) [inline test/unit/platforms/os_detect_test.rb]: Hah! I just did this too.

File: lib/train/platforms/detect/specifications/os.rb (hunk 2 above)
Comment (mattray) [reply lib/train/platforms/detect/specifications/os.rb]: Updated in latest PR.

File: lib/train/platforms/detect/specifications/os.rb (hunk 1 above)
Comment (mattray) [reply lib/train/platforms/detect/specifications/os.rb]: Updated in latest PR

**Pull request merged**